controllers/utils/multicloud_util.go

```diff
@@ -306,7 +308,7 @@ func DeleteManagedClusterViews(
 	ctx context.Context, c client.Client, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, clusterName string) error {
 
 	var labels = map[string]string{
-		"openshift-cluster-group-upgrades/clusterGroupUpgrade": clusterGroupUpgrade.Namespace + "-" + clusterGroupUpgrade.Name}
+		"openshift-cluster-group-upgrades/clusterGroupUpgrade": clusterGroupUpgrade.Name}
```

Comment:
Missing "openshift-cluster-group-upgrades/clusterGroupUpgradeNamespace". It's possible a cluster is added in two CGUs with same names but different namespaces.